common/transport/amqp/src/amqp.ts

```diff
@@ -145,26 +118,72 @@ export class Amqp {
           putToken: (audience, token, callback) => callback(new errors.NotConnectedError()),
         },
         connecting: {
-          _onEnter: (config, connectCallback) => {
-            let connectError = null;
-            const connectErrorHandler = (err) => {
-              connectError = err;
+          _onEnter: (connectionParameters, connectCallback) => {
+            this._amqpContainer.options.sender_options = {properties: {'com.microsoft:client-version': this._config.userAgentString}, reconnect: false};
+            this._amqpContainer.options.receiver_options = {properties: {'com.microsoft:client-version': this._config.userAgentString}, reconnect: false, autoaccept: autoSettleMode};
+            let manageCreateConnectionHandlers: (operation: string) => void;
+            const createConnectionErrorHandler = (err) => {
+              debug('In the error handler for the connect');
+              manageCreateConnectionHandlers('removeListener');
+              this._fsm.transition('disconnected', connectCallback, err);
+            };
+            const createConnectionDisconnectHandler = (err) => {
+              debug('In the disconnect handler for the connect');
+              manageCreateConnectionHandlers('removeListener');
+              this._fsm.transition('disconnected', connectCallback, err);
             };
+            const createConnectionCloseHandler = (context) => {
+              debug('In the close handler for connect');
+              manageCreateConnectionHandlers('removeListener');
+              this._fsm.transition('disconnected', connectCallback, context);
+            };
+            const createConnectionOpenHandler = (context) => {
+              debug('In the open handler for connect');
+              manageCreateConnectionHandlers('removeListener');
+              this._amqpConnection = context.connection;
+              manageConnectedConnectionHandlers('on');
+              this._fsm.transition('connecting_session', connectCallback, context);
+            };
+            manageCreateConnectionHandlers = (operation: string) => {
+              this._amqpContainer[operation]('connection_open', createConnectionOpenHandler);
+              this._amqpContainer[operation]('connection_close', createConnectionCloseHandler);
+              this._amqpContainer[operation]('error', createConnectionErrorHandler);
+              this._amqpContainer[operation]('disconnected', createConnectionDisconnectHandler);
+            };
+            manageCreateConnectionHandlers('on');
```

Comment:
The intent of this part of the code is hard to guess and the code path even harder to follow. maybe rubber-ducking it would help with finding more descriptive names and comments?